### PR TITLE
Feature/add mimikatz

### PR DESCRIPTION
**Pull request by itaymmguardicore**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -188,3 +189,4 @@ Dependency | License |
  winbind | GPL-3 
  pyinstaller | GPL 
  Celery | BSD 
+ mimikatz | CC BY 4.0 | We use an altered version of mimikatz. Original: https://github.com/gentilkiwi/mimikatz
```
Comment (danielguardicore) [inline README.md]: Update link with our fork.

File: chaos_monkey/config.py [hunk 2]
```diff
@@ -204,17 +204,13 @@ def as_dict(self):
...
-    psexec_passwords = ["Password1!", "1234", "password", "12345678"]
-
-    # ssh exploiter
-    ssh_users = ["root", 'user']
-    ssh_passwords = ["Password1!", "1234", "password", "12345678"]
-
     # rdp exploiter
     rdp_use_vbs_download = True
 
+    # User and password dictionaries for exploits.
```
Comment (danielguardicore) [inline chaos_monkey/config.py]: Add back in the existing defaults

File: chaos_monkey/example.conf [hunk 3]
```diff
@@ -74,11 +58,9 @@
     "serialize_config": false,
     "singleton_mutex_name": "{2384ec59-0df8-4ab9-918c-843740924a28}",
     "skip_exploit_if_file_exist": true,
-    "ssh_users": [
-        "root",
-        "user"
-    ],
     "local_network_scan": true,
+    "exploit_user_list": [],
```
Comment (danielguardicore) [inline chaos_monkey/example.conf]: Add back in the existing defaults

File: chaos_monkey/config.py [hunk 4]
```diff
@@ -204,17 +204,13 @@ def as_dict(self):
...
+    exploit_user_list = []
+    exploit_password_list = []
+
```
Comment (danielguardicore) [inline chaos_monkey/config.py]: Add in a properties to generate user/password pairs, duplicate code in rdpgrinder, smbexec, etc.

File: chaos_monkey/exploit/win_ms08_067.py [hunk 5]
```diff
@@ -233,9 +233,10 @@ def exploit_host(self, host, depth=-1, src_path=None):
                                               src_path,
                                               self._config.dropper_target_path)
 
+        # TODO: why are we doing this? Isn't that smbexec's job?
```
Comment (danielguardicore) [inline chaos_monkey/exploit/win_ms08_067.py]: Could refactor them, currently smbexec only knows to exploit from "nothing"

File: chaos_monkey/system_info/mimikatz_collector.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        except StandardError as ex:
+            LOG.exception("Error getting logon info")
+            return {}
+
+    class LogonData(ctypes.Structure):
+        """
+        Logon data structure returned from mimikatz.
+        """
+        _fields_ = \
+            [
```
Comment (danielguardicore) [inline chaos_monkey/system_info/mimikatz_collector.py]: Name magic constants, same name as in mimikatz

File: monkey_island/cc/main.py [hunk 7]
```diff
@@ -48,6 +48,9 @@
     },
 ]
 
+INITIAL_USERNAMES = ['Administrator', 'root', 'user']
```
Comment (danielguardicore) [inline monkey_island/cc/main.py]: I'd just add this to the default mongodb configuration (same as alive:true), but that's a separate PR/Issue.

File: chaos_monkey/config.py (hunk 4 above)
Comment (itaymmguardicore) [inline chaos_monkey/config.py]: fixed

File: chaos_monkey/config.py (hunk 2 above)
Comment (itaymmguardicore) [inline chaos_monkey/config.py]: fixed

File: chaos_monkey/example.conf (hunk 3 above)
Comment (itaymmguardicore) [inline chaos_monkey/example.conf]: fixed

File: chaos_monkey/system_info/mimikatz_collector.py (hunk 6 above)
Comment (itaymmguardicore) [inline chaos_monkey/system_info/mimikatz_collector.py]: fixed

File: monkey_island/cc/main.py (hunk 7 above)
Comment (itaymmguardicore) [inline monkey_island/cc/main.py]: So it's a future feature 😄 
for now this'll do

File: README.md (hunk 1 above)
Comment (itaymmguardicore) [inline README.md]: fixed

File: chaos_monkey/exploit/win_ms08_067.py (hunk 5 above)
Comment (itaymmguardicore) [inline chaos_monkey/exploit/win_ms08_067.py]: fixed

**Pull request merged**